packages/editor-ui/src/modules/nodeTypes.ts

```diff
@@ -15,64 +27,101 @@ const module: Module<INodeTypesState, IRootState> = {
 		allNodeTypes: (state): INodeTypeDescription[] => {
 			return Object.values(state.nodeTypes);
 		},
-		nodeType: (state) => (nodeType: string, version?: number): INodeTypeDescription | null => {
-			const foundType = Object.values(state.nodeTypes).find(typeData => {
-				const typeVersion = Array.isArray(typeData.version)
-					? typeData.version
-					: [typeData.version];
-
-				return typeData.name === nodeType && typeVersion.includes(
-					version || typeData.defaultVersion || DEFAULT_NODETYPE_VERSION,
-				);
-			});
+		getNodeType: (state) => (nodeTypeName: string, version?: number): INodeTypeDescription | null => {
+			const nodeType = state.nodeTypes[nodeTypeName];
 
-			if (foundType === undefined) {
-				return null;
-			}
+			if (!nodeType) return null;
 
-			return foundType;
+			if (!hasValidVersion(nodeType, version)) return null;
+
+			return nodeType;
 		},
 	},
 	mutations: {
 		setNodeTypes(state, nodeTypesArray: INodeTypeDescription[]) {
-			Vue.set(state, 'nodeTypes', toNodeTypesState(nodeTypesArray));
+			state.nodeTypes = toNodeTypesState(nodeTypesArray);
 		},
 
-		updateNodeTypes(state, nodeTypes: INodeTypeDescription[]) {
-			const oldNodesToKeep = Object.values(state.nodeTypes).filter(
-				node => !nodeTypes.find(
-					n => n.name === node.name && n.version.toString() === node.version.toString(),
-				),
+		updateNodeTypes(state, newNodeTypes: INodeTypeDescription[]) {
+			const oldNodesToKeep = newNodeTypes.reduce(
+				(oldNodes, newNodeType) => omit(newNodeType.name, oldNodes),
+				state.nodeTypes,
 			);
 
-			const newNodesState = [...oldNodesToKeep, ...nodeTypes];
-
-			Vue.set(state, 'nodeTypes', newNodesState);
-
-			state.nodeTypes = toNodeTypesState(newNodesState);
+			state.nodeTypes = { ...oldNodesToKeep, ...toNodeTypesState(newNodeTypes) };
 		},
 
-		removeNodeTypes(state, nodeTypes: INodeTypeDescription[]) {
-			console.log('Store will remove nodes: ', nodeTypes); // eslint-disable-line no-console
+		removeNodeTypes(state, newNodeTypes: INodeTypeDescription[]) {
```

Comment:
nitpick - name does not make sense here
```suggestion
		removeNodeTypes(state, nodeTypesToDelete: INodeTypeDescription[]) {
```